src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -38,59 +38,49 @@ internal interface IAnalyzerAssemblyLoaderInternal : IAnalyzerAssemblyLoader, ID
     /// This type generally assumes that files on disk aren't changing, since it ensure that two calls to <see cref="LoadFromPath(string)"/>
     /// will always return the same thing, per that interface's contract.
     /// </remarks>
-    internal abstract partial class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoaderInternal
+    internal sealed partial class AnalyzerAssemblyLoader : IAnalyzerAssemblyLoaderInternal
     {
         private readonly object _guard = new();
 
         /// <summary>
-        /// Set of analyzer dependencies original full paths to the data calculated for that path
+        /// This is a map between the original full path and how it is represented in this loader.
         /// </summary>
         /// <remarks>
         /// Access must be guarded by <see cref="_guard"/>
         /// </remarks>
-        private readonly Dictionary<string, (AssemblyName? AssemblyName, string RealAssemblyPath)?> _analyzerAssemblyInfoMap = new();
+        private readonly Dictionary<string, (IAnalyzerPathResolver? Resolver, string RealPath, AssemblyName? AssemblyName)> _originalPathInfoMap = new();
```

Comment:
intentioanl that there is no comparer passed here?  I would at least have expected Ordianl (to match the prior files) or OrdinalIgnoreCase (to match FS).  

Making all the dictionaries be explicit in this layer would be helpful.  Along with docs as to why the provided comparer is the right one.

If a comparer is *not* needed/desired, documentation explaining why would also be helpful.  Like "It is intentional that no comparer was provided here as the default string comparison (... which is ...) is the correct semantics we want in this scenario".

Reply:
Added explicit comparers and put rationale into the remarks tags.